packages/syntax-core/src/Tabs/TabInternal.tsx

```diff
@@ -0,0 +1,106 @@
+import { type ComponentProps, type ReactElement, type ReactNode } from "react";
+import classnames from "classnames";
+import Box from "../Box/Box";
+import Typography from "../Typography/Typography";
+import styles from "./TabInternal.module.css";
+
+const ItemCountIcon = ({
+  itemCount,
+  on,
+  selected,
+}: {
+  itemCount: number;
+  on: ComponentProps<typeof TabInternal>["on"];
+  selected: boolean;
+}) => {
+  const unselectedItemCountStyle =
+    on === "lightBackground"
+      ? styles.unselectedItemCount
+      : styles.unselectedItemCountOnDarkBackground;
+  const selectedItemCountStyle =
+    on === "lightBackground"
+      ? styles.selectedItemCount
+      : styles.selectedItemCountOnDarkBackground;
+  const itemCountStyles = classnames(styles.itemCount, {
+    [unselectedItemCountStyle]: !selected,
+    [selectedItemCountStyle]: selected,
```

Comment:
not sure what you prefer, but you can probably just combine those into one big classnames if you want.

```
  const itemCountStyles = classnames(styles.itemCount, {
    [styles.unselectedItemCount]: !selected && on==="lightBackground",
    [styles.unselectedItemCount]: !selected && on!=="lightBackground",
        ... etc
  })
```